docs/source/distribution_implementation.md

```diff
@@ -0,0 +1,452 @@
+      domain=R,
+      paramdomains={"mu": R, "sigma": Rplus},
+      scipy_logcdf=lambda value, mu, sigma: sp.norm.logcdf(value, mu, sigma),
+      decimal=select_by_precision(float64=6, float32=1),
+      n_samples=-1,
+  )
+
+```
+
+These methods will perform a grid evaluation on the combinations of domain and
+paramdomains values, and check that the pymc3 methods and the reference
+functions match. There are a couple of details worth keeping in mind:
+
+1. By default the first and last values (edges) of the `Domain` are not compared
+   (they are used for other things). If it is important to test the edge of the
+   `Domain`, the edge values can be repeated. This is done by the `Bool`:
+   `Bool = Domain([0, 0, 1, 1], "int64")`
+1. There are some default domains (such as `R` and `Rplus`) that you can use for
+   testing your new distribution, but it's also perfectly fine to create your
+   own domains inside the test function if there is a good reason for it (e.g.,
+   when the default values lead too many extreme unlikely combinations that are
+   not very informative about the correctness of the implementation).
+1. By default, a random subset of 100 `param` * `paramdomain` combinations is
+   tested, in order to keep the test runtime under control. When testing your
+   shiny new distribution, you can temporarily set `n_samples=-1` to force all
+   combinations to be tested. This is important to avoid the your `PR` leading
+   to surprising failures in future runs whenever some bad combinations of
+   parameters are randomly tested.
+1. On Github all tests are run twice, under the `aesara.config.floatX` flags of
+   `"float64"` and `"float32"`. However, the reference Python functions will
+   run in a pure "float64" environment, which means the reference and the `Pymc3`
+   results can diverge quite a lot (e.g., underflowing to `-np.inf` for extreme
+   parameters). You should therefore make sure you test locally in both regimes.
+   A quick and dirty way of doing this is to temporariliy add
+   `aesara.config.floatX = "float32"` at the very top of file, immediately after
+   `import aesara`. Remember to set `n_samples=-1` as well to test all
+   combinations. The test output will show what exact parameter values lead to
+   a failure. If you are confident that your implementation is correct, you may
+   opt to tweak the decimal precision with `select_by_precision`, or adjust the
+   tested `Domain` values. In extreme cases, you can mark the test with a
+   conditional `xfail` (if only one of the sub-methods is failing, they should
+   be separated, so that the `xfail` is as narrow as possible):
+
+```python
+
+test_blah_logp(self):
+    ...
+
+@pytest.mark.xfail(
+   condition=(aesara.config.floatX == "float32"),
+   reason="Fails on float32 due to numerical issues",
+)
+test_blah_logcdf(self):
+    ...
+
+
+```
+
+
+## 5. Documenting the new `Distribution`
+
+TODO: What needs to be done? Adding a new entry to the .rst?
```

Comment:
At the very least, they should have a docstring written and then be referenced in the respective API page. Depending on the distribution and it's uses it should probably get a notebook example where it's used in a case study or a tutorial if it has significant difference all other distributions.